datafusion/optimizer/src/analyzer/rewrite_expr.rs

```diff
@@ -319,3 +320,67 @@ fn rewrite_array_concat_operator_to_func_for_column(
         _ => Ok(None),
     }
 }
+
+#[derive(Default)]
+pub struct FunctionSimplifiction {}
+
+impl FunctionSimplifiction {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl AnalyzerRule for FunctionSimplifiction {
+    fn name(&self) -> &str {
+        "function_simplification"
+    }
+
+    fn analyze(&self, plan: LogicalPlan, _: &ConfigOptions) -> Result<LogicalPlan> {
+        function_simplication_analyze_internal(&plan)
+    }
+}
+
+fn function_simplication_analyze_internal(plan: &LogicalPlan) -> Result<LogicalPlan> {
```

Comment:
port from `operator to function` schema is removed since not used

Reply:
I wonder if there is any reason calling simplify needs to be its own analyzer pass? Can't it be called as part of the existing simplification code?

```
logical_plan after simplify_expressions SAME TEXT AS ABOVE
```

That would also allow this function to be used by anyone else that used the simplification API directly: https://github.com/apache/arrow-datafusion/blob/8f3d1ef23f93cd4303745eba76c0850b39774d07/datafusion-examples/examples/expr_api.rs#L113

Reply:
No special reason. I forgot this analyzer exists. Let me take a look